src/v3/src/components/ChromeDtcContainer/ChromeDtcContainer.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { h } from 'preact';
+
+import { useWidgetContext } from '../../contexts';
+import {
+  ChromeDtcContainerElement, UISchemaElementComponent,
+} from '../../types';
+import { getBaseUrl } from '../../util';
+import HiddenIFrame from '../HiddenIFrame';
+
+const ChromeDtcContainer: UISchemaElementComponent<{
+  uischema: ChromeDtcContainerElement
+}> = ({ uischema }) => {
+  const {
+    options: {
+      href,
+    },
+  } = uischema;
+  const { widgetProps } = useWidgetContext();
+  const deviceChallengeUrl = new URL(href as string, getBaseUrl(widgetProps)).toString();
```

Comment:
is the `as string` cast needed? `href` should already be `string`

Reply:
also just wondering if the `new URL` part is strictly necessary to get the full URL for the iframe. I just don't see this being done in gen2 but it also uses the courage `View` which might do stuff like this for us. If we do need the fully qualified URL then using this is the right way to do it (as opposed to string concatentation)

Reply:
This URL construction logic was copied over from how `OpenOktaVerifyFPButton.tsx` constructs its URL, which was added via Lee's PR here: https://github.com/okta/okta-signin-widget/pull/3031

You're right that the `as string` cast is not necessary here because I made `href` required for the `ChromeDtcContainerElement` `options`, good callout. In terms of the constructing the URL via a URL object, I think for consistency's sake it's fine to leave it since that's how the button component constructs the URL, unless you have strong opinions

Reply:
> In terms of the constructing the URL via a URL object, I think for consistency's sake it's fine to leave it since that's how the button component constructs the URL, unless you have strong opinions

agree. let's keep it then, thanks for the explanation